### PR TITLE
chore: locale autodetection

### LINKED ISSUE #31: Setup locale autodetection
Setup lingui locale detection with default, storage and URL (lang query) method. Make sure language switcher persists the changes across reloads.

Also, make sure the storage key is compatible with ember site so that it picks up any change in locale from localstorage and vice versa

### PR DESCRIPTION
**Pull request by pc-beast**
Fixes #31

### CODE REVIEW

File: utils/i18n.ts [hunk 1]
```diff
@@ -14,3 +14,10 @@ export async function activate(locale: string): Promise<void> {
   i18n.load(locale, messages)
   i18n.activate(locale)
 }
+
+export async function activateAndSetCookie(locale: string): Promise<void> {
+  const { messages } = await import(`../locale/${locale}/messages.js`)
+  i18n.load(locale, messages)
+  i18n.activate(locale)
+  document.cookie = `current_locale=${locale}`
```
Comment (takshch) [inline utils/i18n.ts:22]: I think We should not use `document.cookie`. It can cause Security Issues later.

Comment (pc-beast) [reply utils/i18n.ts:22]: old site uses cookies for saving locale

Comment (takshch) [reply utils/i18n.ts:22]: Can you please tell me why old site uses cookies for saving locale?

File: pages/_app.tsx [hunk 2]
```diff
@@ -35,6 +36,13 @@ function MyApp({
     }
     // Activate translation on client side after locale detection
     // TODO: To be implemented
+    const DEFAULT_FALLBACK = () => 'en'
+    const locale = detect(
+      fromUrl('lang'),
+      fromCookie('current_locale'),
```
Comment (iamareebjamal) [inline pages/_app.tsx:42]: fromNavigator after this

File: pages/_app.tsx [hunk 3]
```diff
@@ -35,6 +36,13 @@ function MyApp({
...
+      DEFAULT_FALLBACK
+    )
+    activateAndSetCookie(locale)
```
Comment (iamareebjamal) [inline pages/_app.tsx:45]: Move this in lingui file as well and name the function `detectAndSetLocale`

File: utils/i18n.ts [hunk 4]
```diff
@@ -14,3 +14,10 @@ export async function activate(locale: string): Promise<void> {
   i18n.load(locale, messages)
   i18n.activate(locale)
 }
+
+export async function activateAndSetCookie(locale: string): Promise<void> {
+  const { messages } = await import(`../locale/${locale}/messages.js`)
+  i18n.load(locale, messages)
+  i18n.activate(locale)
```
Comment (iamareebjamal) [inline utils/i18n.ts:21]: Reuse the activate function above. Don't repeat code

File: utils/i18n.ts [hunk 5]
```diff
@@ -16,8 +22,17 @@ export async function activate(locale: string): Promise<void> {
 }
 
 export async function activateAndSetCookie(locale: string): Promise<void> {
-  const { messages } = await import(`../locale/${locale}/messages.js`)
-  i18n.load(locale, messages)
-  i18n.activate(locale)
+  activate(locale)
```
Comment (iamareebjamal) [inline utils/i18n.ts:25]: You need to await this

File: utils/i18n.ts [hunk 6]
```diff
@@ -16,8 +22,17 @@ export async function activate(locale: string): Promise<void> {
...
+
+export async function detectAndSetLocale(): Promise<void> {
+  const DEFAULT_FALLBACK = () => 'en'
+  const locale = detect(
+    fromUrl('lang'),
+    fromCookie('current_locale'),
+    fromNavigator(),
+    DEFAULT_FALLBACK
+  )
+  activateAndSetCookie(locale)
```
Comment (iamareebjamal) [inline utils/i18n.ts:37]: await

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**